conformance/lang/expressions/type_cast_expression.balt

```diff
@@ -0,0 +1,1698 @@
+Test-Case: parser-error
+Description: Test syntax errors in type cast expression.
+Labels: any, type-cast-expr
+
+function errorFunction() {
+    any a = 4;
+    _ = int> a; // @error operator not defined for typedesc int an any
+
+    _ = <int a; // @error missing GT token
+
+    _ = <> a; // @error missing type descriptor
+
+    _ = <int> ; // @error missing expression
+
+    _ = <int %> a; // @error invalid token `%`
+
+    _ = <$int> a; // @error invalid token `$`
+}
+
+Test-Case: output
+Description: Test type cast param containing annotations.
+Labels: annotation, annotation-decl, any, map-type, mapping-constructor-expr, string, type-cast-expr
+
+annotation map<string> myAnnot on type;
+
+function init() {
+    any a = 5;
+
+    any b = <@myAnnot {name: "Ballerina"}>a;
+    io:println(b); // @output 5
+
+    any c = <@untainted>a;
+    io:println(c); // @output 5
+
+    any d = <@deprecated>a;
+    io:println(d); // @output 5
+
+    int e = <@myAnnot {name: "Ballerina"} int>a;
```

Comment:
In order to cover

> However, it is also allowed for the parameter to consist only of annotations; in this case, the only effect of the type cast is for the contextually expected type for expression to be augmented with the specified annotations.

shall we move the tests with only annotations to a separate test?